asTime fixed for time-only values

bizley opened the pull request:
| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #13343

Commit:
asTime fixed for time-only values

Commit:
Merge branch 'master' into 13343-formatter-time-only-values

cebe commented on framework/i18n/Formatter.php:
```diff
@@ -650,40 +650,46 @@ private function formatDateTimeValue($value, $format, $type)
      * Since version 2.0.1 this may also return an array if `$checkTimeInfo` is true.
      * The first element of the array is the normalized timestamp and the second is a boolean indicating whether
      * the timestamp has time information or it is just a date value.
+     * Since version 2.0.12 the array has third boolean element indicating whether the timestamp has date information.
```
A note in the `UPGRADE` file is needed.

bizley replied:
`UPGRADE` draft added.

cebe commented on tests/framework/i18n/FormatterDateTest.php:
```diff
@@ -687,6 +687,18 @@ public function testDateOnlyValues()
+        // ensure this does not happen on time only values
+        $this->assertSame('10:32', $this->formatter->asTime('10:32', 'HH:mm'));
+        
+        $this->assertSame('9am', $this->formatter->asTime('9am', 'Ha'));
```
a test including seconds would be nice.

bizley replied:
Done.

Commit:
Merged branch master into master

Commit:
Merged branch master into 13343-formatter-time-only-values

Commit:
Requested changes

- fixed CHANGELOG
- added UPGRADE info
- method doc updated with more info and bool var type instead of boolean
- added test with seconds

dynasource reviewed (changes requested):
could this PR be completed with a unit test which matches the clear use case of https://github.com/yiisoft/yii2/issues/13343#issuecomment-271119689 (including a reference?)

Commit:
Merged branch master into master

Commit:
Merged branch master into 13343-formatter-time-only-values

SilverFire commented:
https://github.com/yiisoft/yii2/pull/13348/files#diff-f1eb247f183bf542c55a57d24506a469R698
Isn't this enough?

bizley commented:
It's ok, I've just finished the exact use case test, commit in few seconds.

Commit:
updated test for reported use case

dynasource reviewed (approved):
thanks!

cebe commented:
Merged, thank you!
